src/System.Net.Primitives/src/System/Net/HostInformationPal.Windows.cs

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Threading;
+
+using Microsoft.Win32.SafeHandles;
+
+namespace System.Net.NetworkInformation
+{
+    internal static class HostInformationPal
+    {
+        // Changing this information requires a reboot, so it's safe to cache.
+        private static Interop.IpHlpApi.FIXED_INFO s_fixedInfo;
+        private static bool s_fixedInfoInitialized;
+        private static object s_syncObject = new object();
+
+        private static Interop.IpHlpApi.FIXED_INFO GetFixedInfo()
+        {
+            uint size = 0;
+            SafeLocalAllocHandle buffer = null;
+            Interop.IpHlpApi.FIXED_INFO fixedInfo = new Interop.IpHlpApi.FIXED_INFO();
+
+            // First we need to get the size of the buffer
+            uint result = Interop.IpHlpApi.GetNetworkParams(SafeLocalAllocHandle.InvalidHandle, ref size);
+
+            while (result == Interop.IpHlpApi.ERROR_BUFFER_OVERFLOW)
+            {
+                try
```

Comment:
I understand this is legacy code, but would it be possible to replace try/finally with using here?